pkg/tools/sandbox/node.go

```diff
@@ -116,27 +118,37 @@ func (n *Node) newEndpoint(
 	nse.Url = u.String()
 
 	// 3. Register with the node registry client
+	err = n.RegisterEndpoint(ctx, nse, registryClient)
+	if err != nil {
+		return nil, err
+	}
 
+	log.FromContext(ctx).Infof("Started listen endpoint %s on %s.", nse.Name, u.String())
+
+	return &EndpointEntry{Endpoint: ep, URL: u}, nil
+}
+
+// RegisterEndpoint - registers endpoint in the registry client
+func (n *Node) RegisterEndpoint(ctx context.Context, nse *registryapi.NetworkServiceEndpoint, registryClient registryapi.NetworkServiceEndpointRegistryClient) error {
```

Comment:
We should possibly have here `RegisterForwarder` and `RegisterEndpoint` without passing registry client.

Reply:
Or we can have `registerEndpoint` with passing registry clients for both forwarders, endpoints to be used internally and `RegisterEndpoint` only for endpoints to be used from tests.

Reply:
Done